hudi-common/src/main/java/org/apache/hudi/common/model/HoodieIndexDefinition.java

```diff
@@ -81,6 +81,10 @@ public String getIndexType() {
     return indexType;
   }
 
+  public String getIndexNameWithoutPrefix() {
+    return indexName.substring(indexType.length() + 1);
```

Comment:
usually the prefix is MetadataPartitionType.getPartitionPath. Should we move the method to that enum?

Reply:
I think a better way is to get MetadataPartitionType inside HoodieIndexDefinition. Moving the function there would not be very clean right now.

Reply:
lets UT (unit test) this method?

Reply:
Changed the method to use MetadataPartitionType itself. Added a UT as well